test_runner/random_test_runner/include/random_test_runner/file_interactions/yaml_test_params_saver.hpp

```diff
@@ -40,11 +40,20 @@ struct convert<TestSuiteParameters>
     node["ego_goal_lanelet_id"] = rhs.ego_goal_lanelet_id;
     node["ego_goal_partial_randomization"] = rhs.ego_goal_partial_randomization;
     node["ego_goal_partial_randomization_distance"] = rhs.ego_goal_partial_randomization_distance;
-    node["npc_count"] = rhs.npcs_count;
-    node["npc_min_speed"] = rhs.npc_min_speed;
-    node["npc_max_speed"] = rhs.npc_max_speed;
-    node["npc_min_spawn_distance_from_ego"] = rhs.npc_min_spawn_distance_from_ego;
-    node["npc_max_spawn_distance_from_ego"] = rhs.npc_max_spawn_distance_from_ego;
+    node["npc_vehicle_count"] = rhs.npc_vehicle_count;
+    node["npc_vehicle_min_speed"] = rhs.npc_vehicle_min_speed;
+    node["npc_vehicle_max_speed"] = rhs.npc_vehicle_max_speed;
+    node["npc_vehicle_min_spawn_distance_from_ego"] = rhs.npc_vehicle_min_spawn_distance_from_ego;
+    node["npc_vehicle_max_spawn_distance_from_ego"] = rhs.npc_vehicle_max_spawn_distance_from_ego;
+    node["npc_pedestrian_count"] = rhs.npc_pedestrian_count;
+    node["npc_pedestrian_planner"] = rhs.npc_pedestrian_planner;
+    node["npc_pedestrian_min_speed"] = rhs.npc_pedestrian_min_speed;
+    node["npc_pedestrian_max_speed"] = rhs.npc_pedestrian_max_speed;
+    node["npc_pedestrian_behavior_static"] = rhs.npc_pedestrian_behavior_static;
+    node["npc_pedestrian_behavior_crosswalk"] = rhs.npc_pedestrian_behavior_crosswalk;
+    node["npc_pedestrian_behavior_walk_along_lane"] = rhs.npc_pedestrian_behavior_walk_along_lane;
+    node["npc_pedestrian_lanelet_min_offset"] = rhs.npc_pedestrian_lanelet_min_offset;
+    node["npc_pedestrian_lanelet_min_offset"] = rhs.npc_pedestrian_lanelet_min_offset;
```

Comment:
The number of parameters is huge and the documentation and implementation points are far apart, which may cause a large discrepancy between the documentation and the implementation.
Instead of defining parameters directly in yaml, please implement them using [generate_parameter_library](https://github.com/PickNikRobotics/generate_parameter_library), which can automatically generate C++ source code for rosparam from yaml.